WooCommerce/Classes/Blaze/BlazeLocalNotificationScheduler.swift

```diff
@@ -177,6 +171,12 @@ private extension DefaultBlazeLocalNotificationScheduler {
     }
 
     func scheduleNoCampaignReminderIfNeeded() {
+        /// Upon logging out, `CoreDataManager` clears the storage triggering data change.
+        /// Checking the authentication state helps avoiding reloading data
+        /// in the unauthenticated state.
+        guard stores.isAuthenticated else {
+            return
+        }
```

Comment:
Moved this here as previously we were checking only when the storage is reset. Now this is needed when the storage is cleared.